### PR TITLE
[FEAT] Add ability to insert distributed columns at pandas query compiler

### LINKED ISSUE #1772: Add ability to insert distributed columns at Pandas query compiler
For now, there is no way to insert one or multiple columns in a dataframe at a specific location if you are working at the query compiler level, without materializing column and passing it into `insert`.

### PR DESCRIPTION
**Pull request by dchigarev**
Signed-off-by: Dmitry Chigarev <dmitry.chigarev@intel.com>

## What do these changes do?

- [ ] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/contributing.html)
- [ ] passes `flake8 modin`
- [ ] passes `black --check modin`
- [ ] signed commit with `git commit -s` 
- [ ] Resolves #1772 
- [ ] tests added and passing

### CODE REVIEW

**Comment by dchigarev** [comment]
@devin-petersohn I've implemented distributed insert for pandas query compiler, it allows to insert multiple columns in a scalable way by broadcasting parts with columns to insert to partitions, that specified in `locs`. This is a draft implementation and I believe that there's a lot of places where it could be optimized, but for now I've got this times:
```
inserting 6 columns with `_insert_qc`: 0.8802017914131284
inserting 6 columns with `insert`: 7.764727934263647
inserting 1 columns with `_insert_qc`: 0.33948098309338093
inserting 1 columns with `insert`: 1.643635481595993
```
<details><summary>Script that measured it</summary>



```python
import numpy as np
import string
import csv
import os
from timeit import default_timer as timer

IS_PANDAS = True
RAND_LOW = -100
RAND_HIGH = 100
N = 5000
M = 128
TEST_FILENAME = os.path.abspath(f"int_dataset-{N},{M},{RAND_LOW},{RAND_HIGH}.csv")


def generate_data_file(filename, row_n, col_n):
    letters = list(string.ascii_letters)
    columns = [
        f"col{i}"
        for i in np.arange(col_n)
    ]

    with open(filename, "w") as file:
        writer = csv.writer(file, delimiter=",")
        writer.writerow(columns)
        for _ in np.arange(row_n):
            writer.writerow(np.random.randint(RAND_LOW, RAND_HIGH, col_n))


if __name__ == "__main__":
    import modin.pandas as pd

    if not os.path.exists(TEST_FILENAME):
        generate_data_file(TEST_FILENAME, N, M)

    md_df = pd.read_csv(TEST_FILENAME)

    cols = ["col0", "col1", "col2", "col50", "col64", "col90"]
    locs = [0, 1, 2, 3, 50, 51]

    def measure_time(cols, locs):
        src = md_df.drop(columns=cols)
        to_insert = md_df[cols]

        t1 = timer()
        result = src._query_compiler._insert_qc(locs, to_insert._query_compiler)
        print(f"inserting {len(locs)} columns with `_insert_qc`:", timer() - t1)

        src = md_df.drop(columns=cols)
        t1 = timer()
        for i in range(len(locs)):
            src.insert(loc=locs[i], column=to_insert.columns[i], value=to_insert[to_insert.columns[i]])
        print(f"inserting {len(locs)} columns with `insert`:", timer() - t1)

    measure_time(cols, locs)
    measure_time(cols[:1], locs[:1])
```


</details>


@devin-petersohn I need your brief review and ideas how it could be integrated at the API level, because in a current implementation all query compilers are supposed to obtain column as a value, not query compiler. But I believe that even at query compiler level this could a useful feature, for example, I'm going to use it in a `pivot_table` implementation (#1669), when inserting multiple columns with margins at the resulting frame.

File: modin/backends/pandas/query_compiler.py [hunk 1]
```diff
@@ -683,7 +683,7 @@ def resample_app_ser(self, resample_args, func, *args, **kwargs):
             df_op=lambda df: df.squeeze(axis=1),
             func=func,
             *args,
-            **kwargs
+            **kwargs,
```
Comment (devin-petersohn) [inline modin/backends/pandas/query_compiler.py:686]: Please move formatting changes to a new PR

File: modin/backends/pandas/query_compiler.py [hunk 2]
```diff
@@ -1788,7 +1853,7 @@ def compute_groupby(df):
             try:
                 agg_func(
                     pandas.DataFrame(index=[1], columns=[1]).groupby(level=0),
-                    **agg_args
+                    **agg_args,
```
Comment (devin-petersohn) [inline modin/backends/pandas/query_compiler.py:1856]: Formatting change in another PR please.

**Comment by dchigarev** [comment]
closing in glory of #2443

**Pull request closed without merging**